tools/Az.Tools.Predictor/Az.Tools.Predictor/ParameterValuePredictor.cs

```diff
@@ -55,18 +60,23 @@ public void ProcessHistoryCommand(CommandAst command)
         /// <returns>The parameter value from the history command. Null if that is not available.</returns>
         public string GetParameterValueFromAzCommand(string commandNoun, string parameterName)
         {
-            if (_command_param_to_resource_map.ContainsKey(commandNoun))
+            parameterName = parameterName.ToLower();
+            var key = parameterName;
+            Dictionary<string, string> commandNounMap = null;
+
+            if (_command_param_to_resource_map?.TryGetValue(commandNoun, out commandNounMap) == true)
```

Comment:
Is this necessary? I'm OK with being explicit, but none of the other ifs are.

Reply:
It is. It's using "?." which returns a "bool?". Cannot use "bool?" directly inside `if`.  The check means _command_param_to_resource_map` is not `null` and it has the key `commandNoun`.

Reply:
Oh I see now, thank you for clarifying.